src/main/java/com/redislabs/redistimeseries/RedisTimeSeries.java

```diff
@@ -551,6 +551,75 @@ public long add(String sourceKey, long timestamp, double value, long retentionTi
       return ranges;
     }
   }
+  
+  /**
+   * TS.GET key
+   * 
+   * @param key
+   * @return
+   */
+  public Value get(String key) {
+    try (Jedis conn = getConnection()) {
+      List<?> touple = sendCommand(conn, Command.GET, SafeEncoder.encode(key)).getObjectMultiBulkReply();
+      if(touple.isEmpty()) {
+        return null;
+      }
+      return new Value((Long)touple.get(0), Double.parseDouble(SafeEncoder.encode((byte[])touple.get(1))));
+    }
+  }
+
+  /**
+   * TS.MGET [WITHLABELS] FILTER filter...
+   * 
+   * @param key
+   * @param withLabels
+   * @param filters
+   * @return
+   */
+  public Range[] mget(boolean withLabels, String... filters) {
+    try (Jedis conn = getConnection()) {
+      byte[][] args = new byte[1 + (withLabels?1:0) + (filters==null?0:filters.length)][];
+      int i=0;     
+
+      if(withLabels) {
+        args[i++] = Keyword.WITHLABELS.getRaw();
+      }
+  
+      args[i++] = Keyword.FILTER.getRaw();
+      if(filters != null) {
+        for(String label : filters) {
+          args[i++] = SafeEncoder.encode(label);  
+        }
+      }
+          
+      List<?> result = sendCommand(conn, Command.MGET, args).getObjectMultiBulkReply();
+      Range[] ranges = new Range[result.size()];
+      for(int j=0; j<ranges.length; ++j) {
+        List<?> series = (List<?>)result.get(j);
+
+        String resKey = SafeEncoder.encode((byte[])series.get(0));
+
+        List<?> resLables = (List<?>)series.get(1);
+        Map<String, String> rangeLabels = new HashMap<>();
+        for(int l=0; l<resLables.size(); ++l) {
+          List<byte[]> label = (List<byte[]>)resLables.get(l);
+          rangeLabels.put( SafeEncoder.encode(label.get(0)), SafeEncoder.encode(label.get(1)));
+        }   
+
+        List<?> touple = (List<?>)series.get(2);
+        Value[] values;
+        if(touple.isEmpty()) {
+          values = new Value[0];
```

Comment:
@gkorland should we address the open PR on makint the reply of `TS.MGET` consistent with `TS.MRANGE` prior approving this one?

https://github.com/RedisTimeSeries/RedisTimeSeries/pull/321

Reply:
After revising this internally we've decided to keep the different reply types . Approving :)